From a0de2b0cbf1d33173642efd43dc45f0a005ab968 Mon Sep 17 00:00:00 2001
From: "M. Vefa Bicakci" <vefa.bicakci@windriver.com>
Date: Thu, 17 Feb 2022 14:17:58 -0500
Subject: [PATCH] driver core: auxiliary bus: Enable by default

This commit enables CONFIG_AUXILIARY_BUS by default. This is necessary,
because StarlingX does not enable any kernel modules that 'select'
CONFIG_AUXILIARY_BUS, which causes the kernel's build system to
automatically disable this option.

However, StarlingX has out-of-tree users of this functionality (such as
the 'ice' and 'mlnx-ofa_kernel' out-of-tree kernel driver packages),
hence the need for this change. Prior to this commit, the aforementioned
out-of-tree kernel drivers would use their embedded/bundled copy of the
same functionality, which would cause kernel symbol conflicts at
run-time when users attempted to load both ice and mlnx-ofa_kernel
drivers.

Signed-off-by: M. Vefa Bicakci <vefa.bicakci@windriver.com>
---
 drivers/base/Kconfig | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/base/Kconfig b/drivers/base/Kconfig
index 6f04b831a..f6177ea14 100644
--- a/drivers/base/Kconfig
+++ b/drivers/base/Kconfig
@@ -3,6 +3,7 @@ menu "Generic Driver Options"
 
 config AUXILIARY_BUS
 	bool
+	default y
 
 config UEVENT_HELPER
 	bool "Support for uevent helper"
-- 
2.17.1

